js/utility/bc-google-analytics.js

```diff
@@ -79,13 +90,106 @@ bcpl.utility.googleAnalytics = (() => {
 		addOutboundLinkTracking();
 	};
 
+	const getDefaultKey = (action) => hasOwnProperty(defaultGoogleEvents, action) ? 
+		defaultGoogleEvents[action] : 
+		null;
+
+	const getDefaultEvent = (event) => {
+		const {
+			action,
+			label
+		} = event;
+		const defaultKey = getDefaultKey(action);
+
+		return defaultKey ? 
+			{
+				[defaultKey]: label
+			} : null;
+	};
+
+	const getStandardEvent = (event) => {
+		const {
+			category,
+			label,
+			value
+		} = event;
+		const eventObj = {};
+
+		if (category) {
+			eventObj[googleEventKeys.category] = category;
+		}
+		if (label) {
+			eventObj[googleEventKeys.label] = label;
+		}
+		if (value) {
+			eventObj[googleEventKeys.value] = value;
+		}
+
+		return eventObj;
+	};
+
+	const getEventData = (event) => getDefaultEvent(event) || getStandardEvent(event);
+
+	const trackEvent = (event) => {
+		const eventData = getEventData(event);
+
+		if (eventData) {
```

Comment:
think there may be an issue with this. In the case that the event is not default, the getStandardEvent would return {} which is truthy. it would probably be worth. It would be worth adding tests for trackEvent to test this scenario. You would need to mock the gtag method.